src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -49,18 +75,47 @@ public void Start()
         /// <param name="timeout">Timeout to wait for the shutdown to complete, in milliseconds.</param>
         public void Shutdown(int timeout)
         {
-            HcsFunctions.ShutdownComputeSystem(_id, timeout);
+            ShutdownAsync();
+            if (!WaitShutdown(timeout))
+            {
+                throw new TimeoutException();
+            }
+        }
+
+        public void ShutdownAsync()
+        {
+            string result;
+            HcsFunctions.ProcessHcsCall(HcsFunctions.HcsShutdownComputeSystem(_cs, null, out result));
+        }
+
+        public bool WaitShutdown(int timeout)
```

Comment:
Yes, but only ShutdownAsync would return the task.  Shutdown would still be here and be synchronous.  Update incoming!